Website: add `@hasStickyFirstColumn` to Advanced Table code docs (#2797)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/specifications/sticky-column.md

## Sticky headers and columns

> The Advanced Table supports setting both the header and the first column as sticky or “pinned” to either the top or left of the container respectively. This can aid in navigation and traversal through large data sets while persisting important identifying values (e.g., names, IDs, etc).

Reviewer: [suggestion] simplifying and cleaning up some of the repetition. 

```suggestion
The Advanced Table supports setting the header and the first column as sticky or “pinned.” This can help users navigate large data sets while persisting important identifying values, such as names and IDs.
```

> There are a few caveats to consider when implementing either of these features:

Reviewer: [suggestion] more plain language

```suggestion
There are a few things to consider when implementing a sticky header or column:
```

> - Instances of the Advanced Table with nested rows, expandable rows, and `colSpan` or `rowSpan` do not support a sticky column because what classifies as the first column is variable depending on these properties.

Reviewer 2: [Question] I know we have a [showcase example](https://hds-showcase-git-hds-4582-advanced-table-stick-f59de9-hashicorp.vercel.app/components/advanced-table#sticky-column-and-nested-rows) of sticky col + nested rows / expand all. So is there a reason for saying it's not supported here? Or is it more guidance that it's not something we encourage our consumers to do?

Reviewer 3: Interesting, I missed that example in the showcase, I wasn't aware that it was even possible. I somewhat see the benefit of keeping the example in the showcase, but I think want to try to discourage this in basically all scenarios because it can result in a pretty broken experience. I can soften the language a bit here if that would be better, but I wonder if there's a more "strict" way to prevent this directly within the component?

Reviewer 2: If we wanted to explicitly prohibit it from the component side there would probably be a way for @shleewhite to do so. If not, we might have to soften the language a tad.

Author: Yeah, I just recently added the showcase example because it is technically possible right now. I can definitely explicitly prohibit it if you prefer @jorytindall

Reviewer 3: For visibility: Lee is going to put some guardrails around implementing a sticky column in tables containing both col/row span and expandable rows. Experience-wise I think this makes sense, if we receive some negative feedback on this or a clear use-case arrises we can reassess in the future.

Author: @dchyun Circling back around, we ended up deciding to explicitly prohibit sticky columns + nested rows - so we can leave the docs as is.

> - The first column is inclusive of the multi-selection column, e.g., setting the first column as sticky in a table with multi-selection will couple the multi-select column and the first column of data together.

Reviewer: [suggestion] simplifying

```suggestion
- Setting the first column as sticky in a table with multi-selection will couple the multi-select column and the first column of data together.
```

![An Advanced Table with a sticky column, where the user has scrolled to showcase the overlapping nature of the column.](/assets/components/table/advanced-table/advanced-table-sticky-column.png)

!!! Info

> At this time only the first column in an Advanced Table can be set as sticky. If you have needs beyond this please [contact](/about/support) the HDS team or [submit a request](https://go.hashi.co/hds-support).

Reviewer: [suggestion] simplifying

```suggestion
We currently only support the first column as sticky. If you have needs beyond this, please [contact](/about/support) the HDS team or [submit a request](https://go.hashi.co/hds-support).
```

!!!

## Overflow

> The Advanced Table supports overflow on both the X and Y axis (horizontally and vertically). For example, when the cumulative width of the columns in the Advanced Table is greater than the width of the viewport or the container the table is in, a subtle shadow is introduced indicating the direction the overflow occurs. When paired with a sticky header or column, the “edge” at which the overflow occurs visually.

Reviewer: [suggestion] simplifying. I'm unsure about the last sentence, but I gave it a shot.

```suggestion
The Advanced Table can overflow on the X and Y axes (horizontally and vertically). When the number of columns no longer fits within the viewport or container, a subtle shadow indicates where the overflow occurs. When paired with a sticky header or column, the “edge” where the overflow occurs uses the same subtle shadow.
```
